hbase-hbtop/src/main/java/org/apache/hadoop/hbase/hbtop/RecordFilter.java

```diff
@@ -0,0 +1,336 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.hbtop;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.hadoop.hbase.classification.InterfaceAudience;
+import org.apache.hadoop.hbase.hbtop.field.Field;
+import org.apache.hadoop.hbase.hbtop.field.FieldValue;
+
+/**
+ * Represents a filter that's filtering the metric {@link Record}s.
+ */
+@InterfaceAudience.Private
+public final class RecordFilter {
+
+  private enum Operator {
+    EQUAL("="),
+    DOUBLE_EQUALS("=="),
+    GREATER(">"),
+    GREATER_OR_EQUAL(">="),
+    LESS("<"),
+    LESS_OR_EQUAL("<=");
+
+    private final String operator;
+
+    Operator(String operator) {
+      this.operator = operator;
+    }
+
+    @Override
+    public String toString() {
+      return operator;
+    }
+  }
+
+  public static RecordFilter parse(String filterString, boolean ignoreCase) {
+    return parse(filterString, Arrays.asList(Field.values()), ignoreCase);
+  }
+
+  public static RecordFilter parse(String filterString, List<Field> fields, boolean ignoreCase) {
+    int index = 0;
```

Comment:
Spent some time to under this method. Two suggestions,
1. A block of comments to make the purpose more clear, it's good to either readability or maintainability.
2. Add a TODO in this method, need a better parsing way.

Reply:
Thanks.

> 1. A block of comments to make the purpose more clear, it's good to either readability or maintainability.

Will do it.

> 2. Add a TODO in this method, need a better parsing way.

Do you have a better idea to parse filter strings? If so, could you please share it with me? And I will rewrite the code for it.

Reply:
> Do you have a better idea to parse filter strings? If so, could you please share it with me? And I can rewrite the code for it.

No, currently. Still having 50 pending-reviewed files waiting for me. So just add a `// TODO` here is good to me. Not in a rush.
Or you can try to come up a new way during the review process.

Reply:
Okay. What is the bad point in the code? Actually not sure that.

Reply:
There're alway better,  like `index-substring` way or `pattern-matching` way?

Reply:
Okay sure. 

Actually, at first, I was thinking of using ANTLR to parse filters but the syntax of filters is very simple so I thought the lib was too much and decided to use a simplest way (the current logic). Similarly, I think  `index-substring` way and `pattern-matching` way are too much in this case, right? If we bring in a more complex syntax of filters to hbtop, we can consider the better way. What do you think?

Reply:
> index-substring way and pattern-matching way are too much in this case, right?

I'm afraid the answer is no, neither is too much. If you have no cycles to polish the code, just add a `// TODO` as commented.

Reply:
Sure. It's not only for branch-1 but also master and branch-2s, so I will file it in a new ticket in Apache Jira. Thanks.

Reply:
I just filed this issue in https://issues.apache.org/jira/browse/HBASE-23064 .

Thanks.